lib/checkclass.cpp

```diff
@@ -562,7 +562,12 @@ void CheckClass::initializeVarList(const Function &func, std::list<const Functio
         }
 
         // Before a new statement there is "[{};()=[]" or ::
-        if (! Token::Match(ftok, "{|}|;|(|)|=|[|::"))
+        // We can also have a new statement after any operators or comparisons
+        if (! Token::Match(ftok, "{|}|;|(|)|=|[|::|%op%|%comp%"))
```

Comment:
It definitely works (at least in this case). I've stepped through the code and the unit tests would be failing if it didn't. I have no problem changing the order though. I also changed the order of the one below this for %oror% as well.